### PR TITLE
deps: from dep to go mod

### PR DESCRIPTION
**Pull request by greut**
- [x] adapt `.travis.yml`
- [x] adapt `Jenkinsfile`
- [x] fix tweak `gometalinter`

### CODE REVIEW

**Comment by greut** [comment]
gometalinter doesn't like go modules as of yet, :sob:

File: .travis.yml [hunk 1]
```diff
@@ -1,30 +1,24 @@
...
+matrix:
+  include:
+  - go: "1.10.x"
+    script:
+      - go test -v ./...
+      - goreleaser --snapshot --skip-sign
+  - go: "1.11.x"
+    script:
+      - go test -v ./...
+      - GO111MODULE=on go build -mod=vendor -o=exoip cmd/exoip/main.go
```
Comment (vincentbernat) [inline .travis.yml]: Why use `-mod=vendor`? We could get caching for free the day we share the cache directory between invocation.

File: Jenkinsfile [hunk 2]
```diff
@@ -47,22 +40,22 @@ node {
...
+def gofmt() {
   docker.withRegistry('https://registry.internal.exoscale.ch') {
-    def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.10')
+    def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.11')
     image.pull()
-    image.inside("-u root --net=host -v ${env.WORKSPACE}/src:/go/src/github.com/${repo}") {
-      sh 'test `gofmt -s -d -e . | tee -a /dev/fd/2 | wc -l` -eq 0'
-      sh "cd /go/src/github.com/${repo} && dep ensure -v -vendor-only"
+    image.inside("-u root --net=host") {
+      sh 'test $(gofmt -s -d -e $(find -iname "*.go" | grep -v "/vendor/") | tee -a /dev/fd/2 | wc -l) -eq 0'
```
Comment (vincentbernat) [inline Jenkinsfile]: Could be just `go fmt ./...`.

File: Jenkinsfile [hunk 3]
```diff
@@ -47,22 +40,22 @@ node {
...
   }
 }
 
 def golint(repo, ...extras) {
   docker.withRegistry('https://registry.internal.exoscale.ch') {
     def image = docker.image('registry.internal.exoscale.ch/exoscale/golang:1.10')
+    image.pull()
     image.inside("-u root --net=host -v ${env.WORKSPACE}/src:/go/src/github.com/${repo}") {
-      sh "golint -set_exit_status -min_confidence 0.6 `go list github.com/${repo}/... | grep -v /vendor/`"
+      sh "golint -set_exit_status -min_confidence 0.3  `go list github.com/${repo}/... | grep -v /vendor/`"
```
Comment (vincentbernat) [inline Jenkinsfile]: Could be `golint -set_exit_status -min_confidence 0.3 ./...` (I think, but nonetheless, it has been several Go versions that `./...` doesn't match vendor anymore).

File: .travis.yml (hunk 1 above)
Comment (vincentbernat) [reply .travis.yml]: OK, I didn't see you committed `vendor/` into VCS.

Comment (greut) [reply .travis.yml]: so I guess the other comments regarding the `grep -v "/vendor/"` are obsolete.

**Comment by greut** [comment]
:cake: that PR was three months old :cake:

**Pull request merged**